old/admin/settings.md
# Server setting / Config

::: {.toctree maxdepth="2" glob=""}
settings/filemanager.rst settings/firewall.rst
:::

## I am not able to login

For installing dependencies we use Composer. As are currently not able
to run it under hestia-php version. We install it via /usr/bin/php. Make
sure proc_open is allowed in the main php version. In the future we look
in methods to allow install via composer via hestia-php.

## Where can I find more information about the config files

A good starting point for every software check the creator

- For Nginx [NGINX Docs](https://nginx.org/en/docs/)
- For Apache2 [Apache Docs](http://httpd.apache.org/docs/2.4/)
- For PHP FPM [PHP
  Docs](https://www.php.net/manual/en/install.fpm.configuration.php)

You could also try [our Forum](https://forum.hestiacp.com)

## Can I use HestiaCP behind Cloudflare CDN?

By default [Cloudflare
Proxy](https://support.cloudflare.com/hc/en-us/articles/200169156-Identifying-network-ports-compatible-with-Cloudflare-s-proxy)
supports only a limited number of ports. This means Cloudflare will not
forward port 8083, which is the default port which HestiaCP listens on.

To change the port to one that Cloudflare will forward, you can run this
command

```bash
v-change-sys-port 2083
```

Or you can disable the proxy feature of Cloudflare

## How to remove unused ethernet ports from RRD

```bash
nano /usr/local/hestia/conf/hestia.conf
```

Add the following line:

```bash
RRD_IFACE_EXCLUDE='lo'
```

Add a other network port as comma separated list

```bash
rm /usr/local/hestia/web/rrd/net/*
systemctl restart hestia
```

## What does the policy "Enforce subdomain ownership" mean

In Hestia 1.3.5 and Vesta it was possible for "User Bob" to create an
subdomain (bob.alice.com) of alice.com even the User Alice is a
different user. This could cause "Security" issues and therefor we
have decided to add a policy to disable / enable this behaviour. By
default the policy is enabled.

To turn the the policy off for only one domain. For example that has
been used for "testing"

```bash
# to enable
v-add-web-domain-allow-users user domain.com
# to disable
v-delete-web-domain-allow-users user domain.com
```

## Can I restrict access to the "Admin" account

In 1.3 we have added the function in Hestia to give an other user
Administrator access for 1.4 we have decided to give system
administrators the options to limit access to the main "system
administrator" account to improve security.

## My server ip has changed what do I need to do

When a server ip has changed run the de following command

```bash
# Update system ip
v-update-sys-ip
```

It will rebuild all config files\...

## Unable to bind adress

In rare cases the network service might be slower than Apache2 and or
Nginx. In that case Nginx or Apache2 will refuse to start up
successfully start.

```bash
systemctl status nginx
```

Will create the error an error

```bash
nginx: [emerg] bind to x.x.x.x:80 failed (99: cannot assign requested address)
```

or in case of Aapche2

```bash
(99)Cannot assign requested address: AH00072: make_sock: could not bind to address x.x.x.x:8443
```

The following command should allow services to assign to non existing ip
addresses

```bash
sysctl -w net.ipv4.ip_nonlocal_bind=1
```

## I am unable to monitor proccess with Zabbix

For security reasons users are not allowed to monitor processes from
other users by default.

To solve the issue in case you use monitoring via Zabbix

Edit /etc/fstab and modify to the following

```bash
proc   /proc   proc   defaults,hidepid=2,gid=zabbix   0   0
```

Reboot the server or remount /proc

## Error: 24: Too many open files

```bash
2022/02/21 15:04:38 [emerg] 51772#51772: open() "/var/log/apache2/domains/<redactedforprivacy>.error.log" failed (24: Too many open files)
```

or

```bash
2022/02/21 15:04:38 [emerg] 2724394#2724394: open() "/var/log/nginx/domains/xxx.error.log" failed (24: Too many open files)
```

This error means that there are to many open files with Nginx. To
resolve this issue:

/etc/systemd/system/nginx.service.d/override.conf

```bash
[Service]
LimitNOFILE=65536
```

Then run:

```bash
systemctl daemon-reload
```

Add this to the Nginx config file (Needs to be smaller or equal to
LimitNOFILE!)

```bash
worker_rlimit_nofile 16384;
```

And then restart nginx with systemctl restart nginx

To verifiy run:

```bash
cat /proc/<nginx-pid>/limits.
```


admin_docs/settings.rst
########################
Server setting / Config
########################

.. toctree::
  :maxdepth: 2
  :glob:
  
  settings/filemanager.rst
  settings/firewall.rst

*********************************************************
I am not able to login 
*********************************************************

For installing dependencies we use Composer. As are currently not able to run it under hestia-php version. We install it via /usr/bin/php. Make sure proc_open is allowed in the main php version. In the future we look in methods to allow install via composer via hestia-php.

*********************************************************
Where can I find more information about the config files
*********************************************************

A good starting point for every software check the creator

* For Nginx `NGINX Docs <https://nginx.org/en/docs/>`_
* For Apache2 `Apache Docs <http://httpd.apache.org/docs/2.4/>`_
* For PHP FPM `PHP Docs <https://www.php.net/manual/en/install.fpm.configuration.php>`_

You could also try `our Forum <https://forum.hestiacp.com>`_


************************************************
Can I use HestiaCP behind Cloudflare CDN?
************************************************

By default  `Cloudflare Proxy <https://support.cloudflare.com/hc/en-us/articles/200169156-Identifying-network-ports-compatible-with-Cloudflare-s-proxy>`_ supports only a limited number of ports. This means Cloudflare will not forward port 8083, which is the default port which HestiaCP listens on.

To change the port to one that Cloudflare will forward, you can run this command

.. code-block:: bash

    v-change-sys-port 2083

Or you can disable the proxy feature of Cloudflare


***************************************************************
How to remove unused ethernet ports from RRD
***************************************************************

.. code-block:: bash

    nano /usr/local/hestia/conf/hestia.conf
    
Add the following line:

.. code-block:: bash

    RRD_IFACE_EXCLUDE='lo'
    
Add a other network port as comma separated list

.. code-block:: bash

    rm /usr/local/hestia/web/rrd/net/*
    systemctl restart hestia
    
***************************************************************
What does the policy "Enforce subdomain ownership" mean
***************************************************************

In Hestia 1.3.5 and Vesta it was possible for "User Bob" to create an subdomain (bob.alice.com) of alice.com even the User Alice is a different user.  This could cause "Security" issues and therefor we have decided to add a policy to disable / enable this behaviour. By default the policy is enabled. 

To turn the the policy off for only one domain. For example that has been used for "testing"

.. code-block:: bash

    # to enable 
    v-add-web-domain-allow-users user domain.com
    # to disable 
    v-delete-web-domain-allow-users user domain.com
    
***************************************************************
Can I restrict access to the "Admin" account
***************************************************************

In 1.3 we have added the function in Hestia to give an other user Administrator  access for 1.4 we have decided to give system administrators the options to limit access to the main "system administrator" account to improve security. 

***************************************************************
My server ip has changed what do I need to do
***************************************************************

When a server ip has changed run the de following command

.. code-block:: bash

    # Update system ip
    v-update-sys-ip
    
It will rebuild all config files...

***************************************************************
Unable to bind adress 
***************************************************************

In rare cases the network service might be slower than Apache2 and or Nginx. In that case Nginx or Apache2 will refuse to start up successfully start.

.. code-block:: bash

  systemctl status nginx
  
Will create the error an error

.. code-block:: bash

    nginx: [emerg] bind to x.x.x.x:80 failed (99: cannot assign requested address)

or in case of Aapche2

.. code-block:: bash

    (99)Cannot assign requested address: AH00072: make_sock: could not bind to address x.x.x.x:8443

 
The following command should allow services to assign to non existing ip addresses

.. code-block:: bash

    sysctl -w net.ipv4.ip_nonlocal_bind=1 

***************************************************************
I am unable to monitor proccess with Zabbix
***************************************************************

For security reasons users are not allowed to monitor processes from other users by default.

To solve the issue in case you use monitoring via Zabbix

Edit /etc/fstab and modify to the following

.. code-block:: bash

  proc   /proc   proc   defaults,hidepid=2,gid=zabbix   0   0
  
Reboot the server or remount /proc 

***************************************************************
Error: 24: Too many open files
***************************************************************

.. code-block:: bash

  2022/02/21 15:04:38 [emerg] 51772#51772: open() "/var/log/apache2/domains/<redactedforprivacy>.error.log" failed (24: Too many open files)
  
or 

.. code-block:: bash

  2022/02/21 15:04:38 [emerg] 2724394#2724394: open() "/var/log/nginx/domains/xxx.error.log" failed (24: Too many open files)
  
This error means that there are to many open files with Nginx. To resolve this issue:

/etc/systemd/system/nginx.service.d/override.conf

.. code-block:: bash 

  [Service]
  LimitNOFILE=65536
  
Then run:

.. code-block:: bash
  
  systemctl daemon-reload
  
Add this to the Nginx config file (Needs to be smaller or equal to LimitNOFILE!)

.. code-block:: bash
  
  worker_rlimit_nofile 16384; 

And then restart nginx with systemctl restart nginx

To verifiy run:

.. code-block:: bash
   
  cat /proc/<nginx-pid>/limits.

release_notes.rst
***************************
Release Notes
***************************

.. toctree::
  :maxdepth: 1
  :glob:

  release_notes/1.6.2.rst
  release_notes/1.6.1.rst
  release_notes/1.6.0.rst
  release_notes/1.5.15.rst
  release_notes/1.5.14.rst
  release_notes/1.5.13.rst
  release_notes/1.5.12.rst
  release_notes/1.5.11.rst
  release_notes/1.5.10.rst
  release_notes/1.5.9.rst
  release_notes/1.5.8.rst
  release_notes/1.5.7.rst
  release_notes/1.5.6.rst
  release_notes/1.5.5.rst
  release_notes/1.5.4.rst
  release_notes/1.5.3.rst
  release_notes/1.5.2.rst
  release_notes/1.5.1.rst
  release_notes/1.5.0.rst
  release_notes/archive


.vitepress/config.js
import { defineConfig } from "vitepress";

export default defineConfig({
	lang: "en-US",
	title: "Hestia Control Panel",
	description: "Open-source web server control panel.",
	head: [
		["link", { rel: "apple-touch-icon", sizes: "180x180", href: "/apple-touch-icon.png" }],
		["link", { rel: "icon", type: "image/png", sizes: "32x32", href: "/favicon-32x32.png" }],
		["link", { rel: "icon", type: "image/png", sizes: "16x16", href: "/favicon-16x16.png" }],
		["link", { rel: "manifest", href: "/site.webmanifest" }],
		["link", { rel: "mask-icon", href: "/safari-pinned-tab.svg", color: "#b7236a" }],
		["meta", { name: "msapplication-TileColor", content: "#b7236a" }],
		["meta", { name: "theme-color", content: "#b7236a" }],
	],

	lastUpdated: true,

	themeConfig: {
		logo: "/logo.png",
		// algolia: {
		//   appId: "2PN8794ZO9",
		//   apiKey: "8a55d7d941cd49b3df09337bc46659c5",
		//   indexName: "hestiacp",
		// },
		nav: nav(),
		socialLinks: [
			{ icon: "github", link: "https://github.com/hestiacp/hestiacp" },
			{ icon: "discord", link: "https://discord.com/invite/yy75DKs" },
			{
				icon: {
					svg: '<svg role="img" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor" stroke-width="2"><title>Forum</title><path stroke-linecap="round" stroke-linejoin="round" d="M17 8h2a2 2 0 012 2v6a2 2 0 01-2 2h-2v4l-4-4H9a1.994 1.994 0 01-1.414-.586m0 0L11 14h4a2 2 0 002-2V6a2 2 0 00-2-2H5a2 2 0 00-2 2v6a2 2 0 002 2h2v4l.586-.586z" /></svg>',
				},
				link: "https://forum.hestiacp.com/",
			},
			{ icon: "twitter", link: "https://twitter.com/HestiaPanel" },
			{ icon: "facebook", link: "https://www.facebook.com/hestiacp" },
		],

		sidebar: { "/docs/": sidebarDocs() },

		editLink: {
			pattern: "https://github.com/hestiacp/hestiacp-docs/edit/main/docs/:path",
			text: "Edit this page on GitHub",
		},
		footer: {
			message: "Released under the GPLv3 License.",
			copyright: "Copyright © 2019-present Hestia Control Panel",
		},
	},
});

/**
 * @returns {import("vitepress").DefaultTheme.NavItem[]}
 */
function nav() {
	return [
		{ text: "Features", link: "/features.md" },
		{ text: "Docs", link: "/docs/getting-started.md", activeMatch: "/docs/" },
		{ text: "Team", link: "/team.md" },
		{ text: "Demo", link: "https://demo.hestiacp.com:8083/" },
		{ text: "Donate", link: "https://www.paypal.com/donate/?cmd=_s-xclick&hosted_button_id=ST87LQH2CHGLA" },
		{
			text: `v1.6.x`,
			items: [
				{
					text: "Changelog",
					link: "https://github.com/hestiacp/hestiacp/blob/main/CHANGELOG.md",
				},
				{
					text: "Contributing",
					link: "https://github.com/hestiacp/hestiacp/blob/main/CONTRIBUTING.md",
				},
				{
					text: "Security policy",
					link: "https://github.com/hestiacp/hestiacp/blob/main/SECURITY.md",
				},
			],
		},
	];
}

/**
 * @returns {import("vitepress").DefaultTheme.SidebarGroup[]}
 */
function sidebarDocs() {
	return [
		{
			text: "Introduction",
			items: [
				{ text: "Getting started", link: "/docs/getting-started" },
				{ text: "Customization", link: "/docs/customization" },
			],
		},
		{
			text: "Admin",
			collapsible: true,
			collapsed: true,
			items: [
				{ text: "API", link: "/docs/admin/api.md" },
				{ text: "API examples", link: "/docs/admin/rest_api.md" },
				{ text: "Backups", link: "/docs/admin/backups.md" },
				{ text: "Cron", link: "/docs/admin/cron.md" },
				{ text: "Database", link: "/docs/admin/database.md" },
				{ text: "PHPmyAdmin SSO", link: "/docs/admin/phpmyadmin-sso.md" },
				{ text: "DNS", link: "/docs/admin/dns.md" },
				{ text: "Mail", link: "/docs/admin/mail.md" },
				{ text: "Settings", link: "/docs/admin/settings.md" },
				{ text: "File manager", link: "/docs/admin/filemanager.md" },
				{ text: "Firewall", link: "/docs/admin/firewall.md" },
				{ text: "Upgrades", link: "/docs/admin/upgrades.md" },
				{ text: "Users", link: "/docs/admin/users.md" },
				{ text: "Web", link: "/docs/admin/web.md" },
				{ text: "FastCGI", link: "/docs/admin/fastcgi.md" },
				{ text: "SSL certificates", link: "/docs/admin/ssl_certificates.md" },
			],
		},
		{
			text: "GUI",
			collapsible: true,
			collapsed: true,
			items: [
				{ text: "Best practices", link: "/docs/gui/best_practices.md" },
				{ text: "User options", link: "/docs/gui/user_options.md" },
				{ text: "Web domains", link: "/docs/gui/webdomains.md" },
			],
		},
		{
			text: "CLI",
			items: [{ text: "Commands", link: "/docs/cli/commands.md" }],
		},
		{
			text: "Contributing",
			items: [
				{ text: "Documentation", link: "/docs/contributing/documentation.md" },
				{ text: "Development", link: "/docs/contributing/development.md" },
				{ text: "Translations", link: "/docs/contributing/translations.md" },
			],
		},
	];
}
